engine/src/main/java/org/hibernate/search/query/dsl/impl/AbstractFacet.java

```diff
@@ -39,9 +41,21 @@ public String getValue() {
 		return value;
 	}
 
+	/**
+	 * @deprecated Use {@link #getFacetFieldName()} instead.
+	 */
 	@Override
+	@Deprecated
 	public String getFieldName() {
-		return fieldName;
+		return getFacetFieldName();
+	}
+
+	public String getFacetFieldName() {
+		return facetFieldName;
+	}
+
+	public String getSourceFieldName() {
```

Comment:
shall we call this _getSourcePropertyName()_ ?

I tend to try disambiguate the entity properties from the Lucene fields by calling them respectivelly "property" and "field" in such cases.